core/unit_test/CMakeLists.txt

```diff
@@ -938,6 +938,6 @@ KOKKOS_ADD_EXECUTABLE_AND_TEST(
   ARGS "one 2 THREE"
 )
 
-if (KOKKOS_ENABLE_HEADER_SELF_CONTAINMENT_TESTS AND NOT KOKKOS_HAS_TRILINOS)
+if (KOKKOS_ENABLE_HEADER_SELF_CONTAINMENT_TESTS AND NOT KOKKOS_HAS_TRILINOS AND NOT WIN32)
```

Comment:
What's the problem here?

Reply:
some includes are missing in windows for whatever reason and it fails inside of xstring with incomplete type issues. Somewhere we include enough of the STL to make that problem go away when you just include Kokkos_Core.hpp, but I don't think we should litter all our headerfiles with more STL includes to make the windows build happy ...